### PR TITLE
Fixed: #9181

### PR DESCRIPTION
**Pull request by sunarcrajneesh**
1. The video is not showing because the @load event does not work with videos in most of browsers. Therefore, we can use 'loadeddata' here.
2. Fix video not focused.

## Issue Reference

#9181 

## Documentation
- [ ] My pull request requires an update on the documentation repository.

### CODE REVIEW

**Comment by ashishkumar-webkul** [comment]
@sunarcrajneesh 

There are some conflicts with the PR please check and fix them.

**Comment by ashishkumar-webkul** [comment]
@sunarcrajneesh 
Please check there are some conflicts in the PR. Please resolve them.

File: packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php [hunk 1]
```diff
@@ -23,20 +23,20 @@ class="flex flex-col max-h-[540px] gap-2.5 [&>*]:flex-[0] overflow-auto scroll-s
                 >
 
                     <img 
-                        :class="`min-w-[100px] max-h-[100px] rounded-xl border transparent cursor-pointer ${activeIndex === index ? 'border border-navyBlue pointer-events-none' : 'border-white'}`"
+                        :class="`min-w-[100px] max-h-[100px] rounded-xl border transparent cursor-pointer ${activeIndex === 'image'+index ? 'border border-navyBlue pointer-events-none' : 'border-white'}`"
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php:26]: Use `'image_' + index` or `image_${index}`. Add underscore.

File: packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php [hunk 2]
```diff
@@ -148,9 +148,11 @@ class="min-w-[450px] max-sm:min-w-full w-[490px]"
     
             mounted() {
                 if (this.media.images.length) {
+                    this.activeIndex = 'image0';
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php:151]: image_0

File: packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php [hunk 3]
```diff
@@ -148,9 +148,11 @@ class="min-w-[450px] max-sm:min-w-full w-[490px]"
...
                     this.baseFile.type = 'image';
                     this.baseFile.path = this.media.images[0].large_image_url;
                 } else if (this.media.videos.length) {
+                    this.activeIndex = 'video0';
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php:155]: video_0

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

File: packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php (hunk 3 above)
Comment (sunarcrajneesh) [reply packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php]: Done

File: packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php [hunk 4]
```diff
@@ -108,96 +108,98 @@ class="min-w-[450px] max-sm:min-w-full w-[490px]"
...
 
-                    if (file.type == 'videos') {
-                        this.baseFile.type = 'video';
-                        this.baseFile.path = file.video_url;
-                        this.isMediaLoading = false;
-                    } else {
-                        this.baseFile.type = 'image';
-                        this.baseFile.path = file.large_image_url;
-                    }
+                    videos: @json(product_video() - > getVideos($product)),
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php:122]: Please check this line, there are extra spaces while chaining.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

File: packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php (hunk 1 above)
Comment (sunarcrajneesh) [reply packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php]: The code has been modified, and String Interpolation has been used.

File: packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php (hunk 4 above)
Comment (sunarcrajneesh) [reply packages/Webkul/Shop/src/Resources/views/products/view/gallery.blade.php]: Formatted

**Review by devansh-webkul — APPROVED** [review]

**Review by ashishkumar-webkul — APPROVED** [review]
Checked video in RTL and LTR view, video is playing, checked in multiple locale in shop front.

**Review by jitendra-webkul — APPROVED** [review]

**Pull request merged**